### ISSUE TITLE
Provide an easy way to mark a response as RESET

### ISSUE THREAD

**Issue by dmoranj** [mcollina/node-coap#43]
This behavior can be currently achieved in the following way: 
```
server.on('request', function (req, res) {
    console.log('This is it: ' + req.payload.toString());
    res._packet.reset = true;
    res._packet.ack = false;
    res.end();
});
```
It seems there isn't any direct way to achieve the same result with the public API. The RESET behavior can be useful to implement some behaviors, as COAP Ping (a RST response to an empty confirmable body, as specified in the COAP specification).

**Comment by mcollina** [comment]
I'm definitely :+1: of having this in this library. Would you like to send a PR to offer a `res.reset()` thing?

On the other end, how this will be exposed to the agent on the client side?

**Comment by dmoranj** [comment]
No problem, I'll try to implement it tomorrow morning. I understand, from the client point of view, a Reset is a response, but with a proper flag indicating is not an ACK, but a RST coming, but I'm not really confident about that (maybe a different event should be emitted?). 

I also doubt if it would be nice to directly implement the COAP Ping mechanism in the server (maybe with a flag to enable/disable it), or if it should be left to implementors of COAP Servers to replicate that behavior in their servers.

**Comment by mcollina** [comment]
I would rather have a module that does the ping logic that on top of this one.

**Comment by mcollina** [comment]
Fixed by #44.
